src/hive/tracer/server.clj

```diff
@@ -1,16 +1,20 @@
 (ns hive.tracer.server
-  (:require [hive.tracer.zmq :as tracer.zmq]
+  (:require [com.stuartsierra.component :as component]
+            [hive.config :as config]
             [hive.tracer.handlers :as tracer.handlers]
-            [com.stuartsierra.component :as component]))
+            [hive.tracer.heartbeat :as tracer.heartbeat]
+            [hive.tracer.zmq :as tracer.zmq]))
 
 (defrecord ZMQServer [port on-receive-fn]
   component/Lifecycle
   (start [this]
     (let [router (tracer.zmq/new-router-socket! port)]
       (assoc this :stop-channel (tracer.zmq/start-receiving! router on-receive-fn)
-                  :router router)))
+             :heartbeat-channel (tracer.heartbeat/start-heartbeat-checking! config/healthcheck-timing-s)
```

Comment:
This heartbeat checking should belong to the store. If you want I can extract and make a component of they

Reply:
I will do it, NP.